lighthouse-cli/random-port.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as http from 'http';
+
+/**
+ * Return a random, unused port.
+ */
+export default function randomPort(): Promise<number> {
```

Comment:
I think this function will return a Promise of `undefined`. Would suggest writing it like:

```
function randomPort() {
  return new Promise((resolve, reject) => {
    const server  = http.createServer();
    server.listen(0, error => {
      if (error) {
        return reject(error);
      }
      const port = server.address().port;
      server.close(error => {
        if (error) {
          return reject(error);
        }
        resolve(port);
      });
    });
  });
}
```